code/game/machinery/status_display.dm

```diff
@@ -148,38 +194,133 @@
 	. = ..()
 
+	// Facing south, we render traditionally.
+	if(dir == SOUTH)
+		add_screen_overlays(.)
+		return
+
+	// Otherwise, we render a projection on the floor.
+
+	// Get screen overlays and return if it's off.
+	var/list/projected_overlays = list()
+	var/anything_displayed = add_screen_overlays(projected_overlays, projection_only = TRUE)
+
+	if(!anything_displayed)
+		return
+
+	// Both of these are vis contents that will receive 1 or more overlays themselves.
+
+	// Matrices to transform the content given the direction
+	var/matrix/floor_matrix = floor_projections["[dir]"]
+	var/matrix/beam_matrix = beam_projections["[dir]"]
+
+	// Make the 2 vis overlays.
+	generate_projection_overlay(floor_matrix, projected_overlays, alpha = PROJECTION_FLOOR_ALPHA)
+	generate_projection_overlay(beam_matrix, projected_overlays, alpha = PROJECTION_BEAM_ALPHA)
+
+	// Impossible for this to look good as an actual emissive since the emissive plane crushes alpha to all or nothing for some reason.
+	// We don't really want anything blocking it anyway since the projection would shine over people.
+	var/mutable_appearance/projection_emissive = mutable_appearance(icon, "projection-mask", offset_spokesman = src, plane = LIGHTING_PLANE)
+	var/list/emissive_offsets = projection_emissive_offsets["[dir]"]
+	projection_emissive.pixel_x = emissive_offsets[1]
+	projection_emissive.pixel_y = emissive_offsets[2]
+	projection_emissive.blend_mode = BLEND_ADD
+	. += projection_emissive
+
+	// Translate the text.
+	message1_overlay?.transform = floor_matrix
+	message1_overlay?.alpha = PROJECTION_TEXT_ALPHA
+	message2_overlay?.transform = floor_matrix
+	message2_overlay?.alpha = PROJECTION_TEXT_ALPHA
+
+/**
+ * Generate a set of vis contents objects for the overlays.
+ *
+ * Needs to be vis contents so that they're mouse transparent.
+ *
+ * Arguments:
+ * * overlay_transform - The matrix transform to apply.
+ * * sub_overlays - The list of sub-overlays to apply.
+ * * alpha - the alpha for the mutable appearance.
+ */
+/obj/machinery/status_display/proc/generate_projection_overlay(matrix/overlay_transform, list/sub_overlays, alpha)
+	PRIVATE_PROC(TRUE)
+
+	var/obj/effect/overlay/new_overlay = SSvis_overlays.add_vis_overlay(
+		src,
+		layer = layer - 0.01, // make sure we're under the text vis objects
+		plane = plane,
+		alpha = alpha,
+		add_appearance_flags = KEEP_TOGETHER,
+		unique = TRUE
+	)
+
+	new_overlay.transform = overlay_transform
+	new_overlay.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	new_overlay.add_overlay(sub_overlays)
+
+	return new_overlay
+
+/**
+ * Generate overlays for the screen. Has an optional offset to shift it around.
+ *
+ * Returns whether the the screen is conceptually on.
+ *
+ * Arguments:
+ * * overlays - the overlays list
+ * * projection_only - only add overlays that are transformed for the projection effect
+ */
+/obj/machinery/status_display/proc/add_screen_overlays(list/overlays, projection_only = FALSE)
```

Comment:
don't call it overlays that's asking for a headache

Reply:
I've changed it to `add_screen_visuals()`.

Reply:
sorry I meant the argument. having two variables, a local one and one stored on /atom be named the same thing can lead to weird errors if you ever rename the arg

Reply:
Oh, haha. Right. I've changed the argument name to `screen_overlays`.